versioned_docs/version-2.0/tutorials/Developers/swe-benchmark.mdx

```diff
@@ -0,0 +1,287 @@
+---
+sidebar_position: 7
+---
+
+# Running SWE-Benchmark with LangSmith
+
+SWE-Benchmark is one of the most popular (and difficult!) benchmarks for developers to test their coding agents against. In this walkthrough we will show you how to load the SWE-benchmark dataset into LangSmith and easily run evals on it, allowing you to have much better visibility into your agents behaviour then using the off-the-shelf SWE-benchmark eval suite. This allows you to pin specific problems quicker and iterate on your agent rapidly to improve performance!
+
+## Loading the data
+
+To load the data, we will pull the `dev` split from Hugging Face, but for your use case you may wish to pull one of the `test`, or `train` splits, and if you want to combine multiple splits you can use `pd.concat`.
+
+```python
+import pandas as pd
+
+splits = {'dev': 'data/dev-00000-of-00001.parquet', 'test': 'data/test-00000-of-00001.parquet', 'train': 'data/train-00000-of-00001.parquet'}
+df = pd.read_parquet("hf://datasets/princeton-nlp/SWE-bench/" + splits["dev"])
+```
+
+### Editing the 'version' column
+
+<div class="alert alert-block alert-danger">
+  This is a very important step! If you skip, the rest of the code WILL NOT
+  WORK!
+</div>
+
+The `version` column contains all string values but all are in float format so they get converted to floats when you upload the CSV to create a LangSmith dataset. Although you can convert the values to strings during your experiments, the issue arises with values like `"0.10"`. When getting converted to a float, you get the value `0.1`, which would become `"0.1"` if you converted it to a string - causing a key error during execution of your proposed patch.
+
+In order to fix this, we need LangSmith to stop trying to convert the `version` column to floats. In order to do this, we can just append a string prefix to each of them that is not float compatible. We then need to split on this prefix when doing evaluation to get the actual `version` value. The prefix we choose here is the string `"version:"`.
+
+<div class="alert alert-block alert-info">
+  The ability to select column types when uploading a CSV to LangSmith will be
+  added in the future to avoid having to use this workaround.
+</div>
+
+```python
+df['version'] = df['version'].apply(lambda x: f"version:{x}")
+```
+
+## Upload the data to LangSmith
+
+### Save to CSV
+
+To upload the data to LangSmith, we first need to save it to a CSV, which we can do using the `to_csv` function provided by pandas. Make sure to save this file somewhere that is easily accesible to you.
+
+```python
+df.to_csv("./../swe-benchmark.csv",index=False)
+```
+
+### Upload CSV to LangSmith
+
+We are now ready to upload the CSV to LangSmith. Once you are on the LangSmith website (smith.langchain.com), go to the `Datasets & Testing` tab on the left side navigation bar, and then click the `+ New Dataset` button in the top right corner.
```

Comment:
I would showcase the sdk for this since they user is already coding, point out the UI approach as an alternative